polyfill/lib/date.mjs

```diff
@@ -139,6 +139,7 @@ export class Date {
     return new MonthDay(GetSlot(this, MONTH), GetSlot(this, DAY));
   }
   static from(arg) {
+    if (typeof arg === 'string') return ES.DateFromString(arg);
     let result = ES.ToDate(arg);
```

Comment:
I think we agreed to call `ToString` for non-object values, but didn't get around to updating the spec to say that. We should probably preserve that behavior. It looks like this changes also causes us to call `ToObject` on non-string values; I don't think that's desirable. Consider:

```js
Number.prototype.year = 2002;
let { year } = 7;
assert(year === 2002);
// And also
assert(Temporal.Date.from(7).year === 2002);
```

Also, the `*FromString` abstract operations in the spec date back to the time before #252, when we also had `Date.fromString`. I'm not sure they still serve a purpose.

Reply:
... and we should have tests.

Reply:
OK, having that context changes things quite a bit :smile: I'll change this back and change the spec instead.

I think this part is pretty well covered by tests (except for checking that non-object arguments are converted to strings, which I'll add.)

I think having `*FromString` as separate abstract operations makes the spec easier to read, but I'm not sure what's usual regarding when to make something an abstract operation. I've just been winging it :smile:

Reply:
Testing that non-object arguments are converted to strings can't currently work because as far as I know there are no non-object, non-string values that would produce a valid ISO string ... until we resolve #198 / #312 when presumably a number like `20200208` would be accepted. I'll add skipped tests for now.